examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,432 @@
+    .rejection_resample(
+        class_func=lambda x: tf.cast(x["TX_FRAUD"], tf.int32),
+        target_dist=[1 - target_rate, target_rate],
+        initial_dist=[1 - fraudulent_rate, fraudulent_rate],
+    )
+    .map(lambda _, x: x)  # Remove the label copy added by "rejection_resample".
+    .batch(batch_size)
+    .map(extract_features_and_label)
+    .prefetch(tf.data.AUTOTUNE)
+)
+
+# The test and validation dataset does not need resampling or shuffling.
+valid_ds = (
+    non_batched_valid_ds.batch(batch_size)
+    .map(extract_features_and_label)
+    .prefetch(tf.data.AUTOTUNE)
+)
+test_ds = (
+    non_batched_test_ds.batch(batch_size)
+    .map(extract_features_and_label)
+    .prefetch(tf.data.AUTOTUNE)
+)
+
+"""
+We print the first four examples of the training dataset. This is a simple way to identify some of the errors that could have been made above.
+"""
+
+for features, labels in train_ds.take(1):
+    print("features")
+    for feature_name, feature_value in features.items():
+        print(f"\t{feature_name}: {feature_value[:4]}")
+    print(f"labels: {labels[:4]}")
+
+"""
+## Train the model
+
+The original dataset is transactional, but the processed data is tabular and only contains normalized numerical values. Therefore, we train a feed-forward neural network.
+"""
+
+inputs = [keras.Input(shape=(1,), name=name) for name in input_feature_names]
+x = keras.layers.concatenate(inputs)
+x = keras.layers.Dense(32, activation="sigmoid")(x)
+x = keras.layers.Dense(16, activation="sigmoid")(x)
+x = keras.layers.Dense(1, activation="sigmoid")(x)
+model = keras.Model(inputs=inputs, outputs=x)
+
+"""
+Our goal is to differentiate between the fraudulent and legitimate transactions, so we use a binary classification objective. Because the dataset is imbalanced, accuracy is not an informative metric. Instead, we evaluate the model using the [area under the curve](https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve) (AUC).
+"""
+
+model.compile(
+    optimizer=keras.optimizers.Adam(0.01),
+    loss=keras.losses.BinaryCrossentropy(),
+    metrics=[keras.metrics.Accuracy(), keras.metrics.AUC()],
+)
+model.fit(train_ds, validation_data=valid_ds)
+
+"""
+We evaluate the model on the test dataset.
+"""
+
+model.evaluate(test_ds)
+
+"""With and AUC of ~83%, our simple fraud detector is showing encouraging results.
+
+
+Plotting the ROC curve is a good solution to understand and select the operation point of the model i.e. the threshold applied on the model output to differentiate between fraudulent and legitimate transactions.
+
+Compute the test predictions:
+"""
+
+predictions = model.predict(test_ds)
+predictions = np.nan_to_num(predictions, nan=0)
+
+"""
+Extract the labels from the test set:
+"""
+
+labels = np.concatenate([label for _, label in test_ds])
+
+"""
+Finaly, we plot the ROC curve.
+"""
+
+_ = RocCurveDisplay.from_predictions(labels, predictions)
+
+"""## Conclusion
```

Comment:
The title should be on the first line *after* `"""`

Reply:
*Done